virttest/utils_net.py

```diff
@@ -3748,23 +3764,27 @@ def get_host_iface():
     return [_.strip() for _ in re.findall("(.*):", host_iface_info)]
 
 
-def get_host_default_gateway(iface_name=False):
+def get_default_gateway(iface_name=False, session=None):
     """
-    Get the Default Gateway or Interface of host.
+    Get the Default Gateway or Interface of host or guest.
 
     :param iface_name: Whether default interface (True), or default gateway
                        (False) is returned.
-    :return: A string of the host's default gateway or interface.
+    :return: A string of the host's or guest's default gateway or interface.
     :rtype: string
     """
     if iface_name:
         cmd = "ip route | awk '/default/ { print $5 }'"
     else:
         cmd = "ip route | awk '/default/ { print $3 }'"
     try:
-        output = decode_to_text(process.system_output(cmd, shell=True).rstrip())
-    except process.CmdError:
-        logging.error("Failed to get the host's default GateWay")
+        if session:
+            output = session.cmd_output(cmd).strip()
+            logging.debug("Guest default gateway is %s" % output)
+        else:
+            output = decode_to_text(process.system_output(cmd, shell=True).rstrip())
+    except (aexpect.ShellError, aexpect.ShellTimeoutError):
```

Comment:
`aexpect.ShellError, aexpect.ShellTimeoutError` can only handle the session part, please add process.CmdError together to handle the host part.

Reply:
Updated, Thank you for the suggestion :)